### ISSUE TITLE
CSS scroll-marker-group modes && WAI-ARIA patterns

### ISSUE THREAD

**Issue by davidsgrogan** [w3c/css-aam#18]
Please take a look at Daniil's [CSS scroll-marker-group modes and WAI-ARIA patterns explainer](https://gist.github.com/danielsakhapov/aa8e744701224994609aebb3e9e316e3).

It describes improving the accessibility of tabs, carousels, and table of contents navigation by having the browser natively manage implicit roles, states, and focus via the `scroll-marker-group` property.

This work was in response to earlier a11y feedback. Any further feedback?

**Comment by spectranaut** [comment]
Discussed today in the ARIA working group: https://www.w3.org/2026/04/16-aria-minutes.html#d106

**Comment by danielsakhapov** [comment]
To try it out - download Chrome Canary and in the URL bar put `chrome://flags/#enable-experimental-web-platform-features`, then enable it (, probably restart) and visit CodePen examples from the explainer ([Tabs mode](https://codepen.io/Daniil-Sakhapov/pen/ogzQQYZ) and [Links mode](https://codepen.io/Daniil-Sakhapov/pen/ZYpmmJE)) or Rob's examples at https://flackr.github.io/web-demos/css-overflow/scroll-marker-type/.
Pay attention to:
- the tab focus order;
- the lack of inactive content in AX tree in Tabs mode;
- how the focus behave when you activate a scroll marker;
- keyboard navigation between the scroll markers in the Tabs mode and lack of such in the Links mode;
- only active scroll marker is a tab stop in the Tabs mode, acting as a focus group, while every scroll marker is a tab stop in the Links mode.

**Comment by jnurthen** [comment]
@aardrian @mcking65 please look at the demo before the meeting on 4/30/2026

**Comment by daniel-montalvo** [comment]
As discussed in 30 April ARIA meeting.

A strange behavior that I am experiencing is that the screen readers focus mode navigation and tab order don't match. Tested with NVDA as of now. To test this:

1. Focus the scroll markers, for example the "introduction" tab by tabbing around, which would disable focus mode.
2. Enable NVDA's focus mode by pressing Insert Space
3. Down arrow.

Expected behavior: screen reader focus mode navigation focuses on "introduction" (h3).

Obtained behavior: Screen reader's focus mode navigation focuses on "Link After the carousel".

**Comment by davidsgrogan** [comment]
@daniel-montalvo, can you help me understand your comment?
https://codepen.io/Daniil-Sakhapov/pen/ogzQQYZ

> Focus the scroll markers, for example the "introduction" tab by tabbing around, which would disable focus mode.

When I tab from "Link before the carousel" to the scroll marker at the bottom, I get put IN to focus mode (I hear the high pitched keyboard sound), even if I was in browse mode.

<img width="610" height="251" alt="Image" src="https://github.com/user-attachments/assets/04f690d1-69a7-4215-b2a0-dddf07c2dce6" />

Then, staying in focus mode, if I hit down arrow, I see focus shift to the second tab marker, which seems correct (down aliased to right):

<img width="732" height="246" alt="Image" src="https://github.com/user-attachments/assets/04d11de5-8067-4f46-a75b-9305e667979a" />

For the sake of argument, perhaps you just swapped the words "browse" and "focus" in your comment? Please confirm if that is the case or not.

If, after focusing the first scroll marker, I switch **to** Browse mode via Insert+Space, then hit Down arrow I _do_ see focus change to "Link after the carousel". But that seems pretty correct? If I start focus at the top, at "Link _before_ the carousel" in Browse mode and hit Down Arrow, I go to "Introduction". So in Browse mode, the arrow keys kind of do spatial navigation? Is that incorrect?

+ @danielsakhapov also: Please correct me if I've misunderstood something

**Comment by daniel-montalvo** [comment]
@davidsgrogan  Thanks for pointing that out, I did swap them. I'll quote from my above comment and correct the issues below.

<blockquote>

1. Focus the scroll markers, for example the "introduction" tab by tabbing around, which would enable focus mode.
1. Enable NVDA's browse mode by pressing Insert Space
1. Down arrow.

Expected behavior: screen reader focus mode navigation focuses on "introduction" (h3, or whatever the content of that scroll marker starts with).

Obtained behavior: Screen reader's focus mode navigation focuses on "Link After the carousel".
Note: This appears to have changed now. In browse mode, you do get to hear all scroll markers, and then you go straight to the link after carousel.

</blockquote>

The main issue I was trying to describe is that when using browse mode, PC cursor, or similar navigation, the scroll markers appear after the content, whereas when you tab around, the scroll markers appear before the content. That seems to me a little bit of a mismatch

**Comment by davidsgrogan** [comment]
Thank you everyone for the feedback raised so far in this issue and during the meetings. Just to close the loop on some and to surface where we're at with others:

(0)
> spectranaut_: I was looking at this this morning and tabbing wasn't working as expected. I'm wondering if I am on the right version
spectranaut_: When I tab, it goes to the other focusable content in the tabs that shouldn't appear
... and the scroll marker doesn't update

We can't reproduce and agree with spectranaut_'s hypothesis that she was probably on the wrong Chrome version. (I see the described behavior when I don't enable `experimental-web-platform-features` in chrome://flags)

(1)
> <Jacques> jamesn: The browser seems to be using the name for either the heading or title of the surrounding div.
<Jacques> jamesn: This seems fine, but I want to understand where it is coming from.
<Jacques> @danielsakhapov : This would be folloing the normal rules.
<Jacques> jamesn: I don't think there is a defined rule for this, it needs to be defined.
<Jacques> jamesn: I'm seeing it in devtools.
<Jacques> jamesn: Its coming from the surrounding div title, so this needs to be defined, and shouldn't be entirely reliant on title.
<Jacques> noah: it looks like this is coming from the surrounding content.
<Jacques> noah: it picks it up from aria-label
<Jacques> jamesn: the name calculation needs to be defined.

From what we can tell, these are following the normal accname rules/algorithm (with some mild assumptions), but jamesn is right that the normal rules aren't defined anywhere to apply to `::scroll-marker` and/or `::scroll-marker-group`. This is a gap that needs to be defined.

(2)
> <Jacques> jamesn: I have other concerns, I have some weird behavior with find on page
<Jacques> ... it switches the tab but the controls don't switch

Fixed in Chrome 152!

(3)
> The main issue I was trying to describe is that when using browse mode, PC cursor, or similar navigation, the scroll markers appear after the content, whereas when you tab around, the scroll markers appear before the content. That seems to me a little bit of a mismatch

Indeed, I see what you're saying. I'm trying to get the history of this to see how intentional it was, and if so, the rationale.
